### PR TITLE
`core:text/edit` Add setup_once and clear_all, clean up old code and add a few comments

### PR DESCRIPTION
**Pull request by Skytrias**
Also fixes the sorted_selection setting the selection implicitly which ruins selection right

### CODE REVIEW

File: core/text/edit/text_edit.odin [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package text_edit
```
Comment (gingerBill) [inline core/text/edit/text_edit.odin:1]: Why was this renamed? `text_edit` is less likely to clash compared to `edit`.

File: core/text/edit/text_edit.odin [hunk 2]
```diff
@@ -107,12 +105,37 @@ end :: proc(s: ^State) {
...
+	s.builder = builder
+	s.selection = { len(builder.buf), 0 }
+	undo_clear(s, &s.undo)
+	undo_clear(s, &s.redo)
 }
 
+// returns true when the builder had content to be cleared
+// clear builder&selection and the undo|redo stacks
+clear_all :: proc(s: ^State) -> (cleared: bool) {
+	if len(s.builder.buf) > 0 {
```
Comment (gingerBill) [inline core/text/edit/text_edit.odin:127]: Maybe check to see if `s.builder` is nil?

**Review by gingerBill — APPROVED** [review]

**Comment by Skytrias** [comment]
alright lovely work on the text_edit bill, would love to see this implemented in other languages this nicely 😄 
my use case right now is `init -> setup_once -> write some text -> clear_all when done` for single edits

**Comment by Skytrias** [comment]
woops sorry for the closing i need to sleep
